marlowe-playground-client/src/Halogen/Blockly.purs

```diff
@@ -147,11 +169,21 @@ handleQuery (GetCodeQuery next) = do
   where
   unexpected s = "An unexpected error has occurred, please raise a support issue at https://github.com/input-output-hk/plutus/issues/new: " <> s
 
-handleAction :: forall m slots. MonadEffect m => Action -> DSL slots m Unit
+handleQuery (HasUnsavedChanges next) = do
+  val <- use _hasUnsavedChanges
+  pure $ Just $ next val
+
+handleQuery (MarkProjectAsSaved next) = do
+  assign _hasUnsavedChanges false
+  pure $ Just $ next
+
+handleAction :: forall m slots. MonadAff m => Action -> DSL slots m Unit
 handleAction (Inject rootBlockName blockDefinitions) = do
   blocklyState <- liftEffect $ Blockly.createBlocklyInstance rootBlockName (ElementId "blocklyWorkspace") (ElementId "blocklyToolbox")
   let
     _ =
+      -- TODO: Refactor blockly to use Effect instead of ST
```

Comment:
I think for this case it's better to leave it as a TODO or change it to something like "This could be refactored to use Effect instead of ST".

It's not a pressing issue that must be resolved, and it can (and probably will) be kept this way for a long time without experience any issues. It is nice though to leave trails of knowledge, so if somebody stumbles upon this code they can see some context.